app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -286,9 +286,9 @@ class BrowserWebViewClient @Inject constructor(
         private val WebBackForwardList.isHttpsUpgrade: Boolean
             get() {
                 if (currentIndex < 1) return false
-                val current = currentItem.originalUrl ?: return false
-                val previous = getItemAtIndex(currentIndex - 1).originalUrl
-                return current == previous.toUri().toHttpsString
+                val current = currentItem.originalUrl?.toUri() ?: return false
```

Comment:
`currentItem` is also `nullable`, so can we go for `currentItem?.originalUrl?.toUri()`